pkg/storaged/stratis-details.jsx

```diff
@@ -59,129 +61,71 @@ function destroy_filesystem(client, fsys) {
-                                 validate: function(disks) {
-                                     if (disks.length === 0)
-                                         return _("At least one disk is needed.");
-                                 },
-                                 spaces: get_available_spaces(client)
-                             })
-            ],
-            Action: {
-                Title: _("Add"),
-                action: function(vals) {
-                    return prepare_available_spaces(client, vals.disks)
-                            .then(paths => {
-                                const devs = paths.map(p => decode_filename(client.blocks[p].PreferredDevice));
-
-                                function add() {
-                                    if (vals.tier == "data") {
-                                        return pool.call("AddDataDevs", [devs])
-                                                .then(([result, code, message]) => {
-                                                    if (code)
-                                                        return Promise.reject(message);
-                                                });
-                                    } else if (vals.tier == "cache") {
-                                        const has_cache = blockdevs.some(bd => bd.Tier == 1);
-                                        const method = has_cache ? "AddCacheDevs" : "InitCache";
-                                        return pool.call(method, [devs])
-                                                .then(([result, code, message]) => {
-                                                    if (code)
-                                                        return Promise.reject(message);
-                                                });
+            dialog_open({
+                Title: _("Add block devices"),
+                Fields: [
+                    SelectOne("tier", _("Tier"),
+                              {
+                                  choices: [
+                                      { value: "data", title: _("Data") },
+                                      { value: "cache", title: _("Cache"), disabled: pool.Encrypted }
+                                  ]
+                              }),
+                    PassInput("passphrase", _("Passphrase"),
+                              {
+                                  visible: () => ask_passphrase,
+                                  validate: val => !val.length && _("Passphrase cannot be empty"),
+                              }),
+                    SelectSpaces("disks", _("Block devices"),
+                                 {
+                                     empty_warning: _("No disks are available."),
+                                     validate: function(disks) {
+                                         if (disks.length === 0)
+                                             return _("At least one disk is needed.");
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18836-20230523-113250-65b80e03-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)